src/components/ui/HomeScreen.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from "react";
+import { FaPlus } from "react-icons/fa";
+import useStore from "../../store";
+
+const DiscoverGrid = () => {
+  const { toggleAddServerModal, connect, isConnecting, connectionError } =
+    useStore();
+  const [query, setQuery] = useState("");
+  const [servers, setServers] = useState<
+    { name: string; description: string; server?: string; port?: string }[]
+  >([]);
+
+  useEffect(() => {
+    const fetchServers = async () => {
+      try {
+        const cachedData = localStorage.getItem("servers");
```

Comment:
I am really creeepied by caching issues, do we really need to cache this instead of feathing every open and reload? 🤔 

Makes things so much simpler to test and live updated...